front-end/src/components/Post/Democracy.tsx

```diff
@@ -0,0 +1,349 @@
+		const convictionValue = event.currentTarget.value as ConvictionType;
+		setConviction(convictionValue);
+	};
+
+	useEffect(() => {
+		async function connect() {
+			const provider = new WsProvider(WS_PROVIDER);
+			const apiResult = await ApiPromise.create({ provider });
+
+			setApi(apiResult);
+			apiResult.isReady.then(() => {
+				setApiReady(true);
+				console.log('API ready');
+			});
+		}
+
+		connect().catch((error) => {
+			queueNotification({
+				header: 'Failed!',
+				message: error.message,
+				status: NotificationStatus.ERROR
+			});
+		});
+	}, []);
+
+	const getLinkedAccount = async (): Promise<InjectedAccountWithMeta | undefined> => {
+		const extensions = await web3Enable(APP);
+
+		if (extensions.length === 0) {
+			queueNotification({
+				header: 'Failed!',
+				message: 'Please install polkadot js extenstion to use this feature',
+				status: NotificationStatus.ERROR
+			});
+
+			return;
+		}
+
+		const accounts = await web3Accounts();
+
+		if (accounts.length === 0) {
+			queueNotification({
+				header: 'Failed!',
+				message: 'Please add accounts to polkadot js extenstion to use this feature',
+				status: NotificationStatus.ERROR
+			});
+
+			return;
+		}
+
+		const accountMap: {[key: string]: InjectedAccountWithMeta} = {};
+
+		accounts.forEach(account => {
+			accountMap[account.address] = account;
+		});
+
+		const linkedAddress = currentUser?.addresses && currentUser?.addresses[0];
+
+		if (!linkedAddress) {
+			queueNotification({
+				header: 'Failed!',
+				message: 'Please link an address in settings',
+				status: NotificationStatus.ERROR
+			});
+
+			return;
+		}
+
+		const linkedAccount = accountMap[linkedAddress || ''];
+
+		if (!linkedAccount) {
+			queueNotification({
+				header: 'Failed!',
+				message: 'Linked account not available',
+				status: NotificationStatus.ERROR
+			});
+
+			return;
+		}
+
+		const injected = await web3FromSource(linkedAccount.meta.source);
+
+		if (api) {
+			api.setSigner(injected.signer);
+		}
+
+		return linkedAccount;
+	};
+
+	const secondProposal = async () => {
+		if (!api) {
+			return;
+		}
+
+		const linkedAccount = await getLinkedAccount();
+
+		if (!linkedAccount) {
+			return;
+		}
+
+		const second = api.tx.democracy.second(onchainId || 0);
+
+		second.signAndSend(linkedAccount.address, ({ status }) => {
+			if (status.isFinalized) {
+				queueNotification({
+					header: 'Success!',
+					message: `Completed at block hash #${status.asFinalized.toString()}`,
+					status: NotificationStatus.SUCCESS
+				});
+			} else {
+				queueNotification({
+					header: 'Info!',
+					message: `Current status: ${status.type}`,
+					status: NotificationStatus.WARNING
+				});
+			}
+		}).catch((error) => {
+			console.log(':( transaction failed');
+			console.error('ERROR:', error);
+			queueNotification({
+				header: 'Failed!',
+				message: error.message,
+				status: NotificationStatus.ERROR
+			});
+		});
+	};
+
+	const voteRefrendum = async (aye: boolean) => {
```

Comment:
This file is huge, can we break it down into at least 3 components with a container and a component for referendum / proposal voting modules?